src/solvers/refinement/string_refinement.cpp

```diff
@@ -33,42 +33,46 @@ Author: Alberto Griggio, alberto.griggio@gmail.com
 #include <langapi/language_util.h>
 #include <java_bytecode/java_types.h>
 
-string_refinementt::string_refinementt(
-  const namespacet &_ns,
-  propt &_prop,
-  unsigned refinement_bound):
-  supert(_ns, _prop),
-  use_counter_example(false),
-  do_concretizing(false),
-  initial_loop_bound(refinement_bound),
-  generator(_ns),
-  non_empty_string(false)
-{ }
-
-/// Add constraints on the size of strings used in the program.
-/// \param i: maximum length which is allowed for strings.
-/// by default the strings length has no other limit
-/// than the maximal integer according to the type of their
-/// length, for instance 2^31-1 for Java.
-void string_refinementt::set_max_string_length(size_t i)
+static bool validate(const string_refinementt::infot &info)
 {
-  generator.max_string_length=i;
+  PRECONDITION(info.ns);
+  PRECONDITION(info.prop);
+  return true;
 }
 
-/// Add constraints on the size of nondet character arrays to ensure they have
-/// length at least 1
-void string_refinementt::enforce_non_empty_string()
+static bv_refinementt::infot bv_refinement_info(
```

Comment:
documentation, for this function and the ones below

Reply:
Local glue function, not sure documentation is necessary.